### DISCUSSION TITLE
[Support]: sync_recordings: true doesn't seem to delete missing/old exports from database - What's the easiest solution?

### DISCUSSION THREAD

**Discussion by distinctjuggle** [blakeblackshear/frigate#14839] [General Support]
### Describe the problem you are having

I've manually deleted all of my video exports from my filesystem since I have created a large number of them via the API working on something else.  Now, Frigate complains about missing files and won't seem to auto-remove them from the database.

I've got:

```
record:
  sync_recordings: true
```

set in my config, but it doesn't seem to auto delete the missing exports when running a simple `docker compose restart frigate`.

Is this intended, or is this a bug?  Does sync_recordings only check recordings against the database and not exports or other files?  I would assume it would check all database values against all files.  

If my assumption is wrong, then is there an easier way for me to delete everything here besides deleting the entire database?  I suppose I could write a program which will auto call the delete export API over all `id`s found from the get exports API, but there really seems like there should be a more elegant built-in Frigate solution.  

### Version

0.15.0 Beta 1

### What browser(s) are you using?

Firefox

### Frigate config file

```yaml
Irrelevant
```

### Relevant Frigate log output

```shell
Irrelevant - it complains about missing files as described of course
```

### Relevant go2rtc log output

```shell
Irrelevant
```

### FFprobe output from your camera

```shell
Irrelevant
```

### Install method

Docker Compose

### docker-compose file or Docker CLI command

```yaml
Irrelevant
```

### Object Detector

Other

### Network connection

Wired

### Camera make and model

Irrelevant

**Comment by NickM-27** [comment]
sync recordings does not touch exports, only the recording segment files

**Reply by distinctjuggle** [reply]
Is there nothing built into Frigate to solve this problem?  I find it hard to believe I'm the first person to ever run into this.

Maybe a similar config value to sync_recordings could be added which looks at other files separately for this same purpose?

**Reply by NickM-27** [reply]
I definitely don't recall anyone ever asking this about exports, typically exports are just managed through the UI. It's possible there could be some button / action in the UI to delete exports that don't exist on the filesystem anymore

**Reply by distinctjuggle** [reply]
I've got way too many to delete manually via UI.  I guess I'll just have to write a program to API delete them all in the meantime.

**Reply by NickM-27** [reply]
it would be simpler to just use `sqlite3 /config/frigate.db` and `DELETE FROM exports;`

**Reply by Iceman248** [reply]
From just doing it, I had to actually use `DELETE FROM "main"."export";` to get it to work.
